src/python/pants/core/util_rules/archive_test.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import tarfile
+import zipfile
+from collections import OrderedDict
+from io import BytesIO
+
+from pants.core.util_rules.archive import ExtractedDigest, MaybeExtractable
+from pants.core.util_rules.archive import rules as archive_rules
+from pants.engine.fs import FileContent, FilesContent, Snapshot
+from pants.engine.rules import RootRule
+from pants.engine.selectors import Params
+from pants.testutil.test_base import TestBase
+
+
+class ArchiveTest(TestBase):
+
+    files = OrderedDict({"foo": b"bar", "hello/world": b"Hello, World!"})
+
+    expected_files_content = FilesContent(
+        [FileContent(name, content) for name, content in files.items()]
+    )
+
+    @classmethod
+    def rules(cls):
+        return (*super().rules(), *archive_rules(), RootRule(Snapshot))
+
+    # TODO: Figure out a way to run these tests without a TestBase subclass, and use
+    #  pytest.mark.parametrize.
+    def _do_test_extract_zip(self, compression) -> None:
+        io = BytesIO()
+        with zipfile.ZipFile(io, "w", compression=compression) as zf:
+            for name, content in self.files.items():
+                zf.writestr(name, content)
+        io.flush()
+        input_snapshot = self.make_snapshot({"test.zip": io.getvalue()})
+        extracted_digest = self.request_single_product(
+            ExtractedDigest, Params(MaybeExtractable(input_snapshot.directory_digest))
+        )
+
+        files_content = self.request_single_product(FilesContent, Params(extracted_digest.digest))
```

Comment:
Ditto on being able to leave off `Params`. Fine to keep if you prefer the consistency, though.

Reply:
I'll leave it, for consistency and easier copypasta in the future.